docs/apis/plugintypes/repository/index.mdx

```diff
@@ -0,0 +1,975 @@
+}
+```
+
+  </TabItem>
+</Tabs>
+
+The return values influence the choices offered to a user when selecting a file in file picker. Consider the screenshot above, which is from the dialogue that appears directly after choosing (but before uploading) a file in mod_resource. The three options result from FILE_INTERNAL, FILE_REFERENCE, and FILE_CONTROLLED_LINK being present. FILE_REFERENCE corresponds to the "alias/shortcut" option.
+
+The option FILE_EXTERNAL is never reflected in the file picker for mod_resource, so its absence or presence in supported_returntypes() is never reflected here. However, FILE_EXTERNAL is the only return type supported by mod_url: For mod_url, file picker will only(!) list repositories that support FILE_EXTERNAL.
+
+This implies that a plugin that uses a file picker is able to narrow the set of supported return types. For example, assignsubmission_file disallows FILE_EXTERNAL and FILE_REFERENCE.
+
+In the end, which type is
+used by Moodle depends on the choices made by the end user (e.g. inserting a link, will result in 'FILE_EXTERNAL'-related functions being used, using a 'shortcut/alias' will result in the 'FILE_REFERENCE'-related functions being used).
+
+#### supported_filetypes()
+
+Optional method that returns the kind of files a repository can store. Returns '\*' for all file types (default implementation), or an array of types or groups (e.g. array('text/plain', 'image/gif', 'web_image')).
+
+<Tabs>
+  <TabItem value="all" label="All files" default>
+
+```php
+function supported_filetypes() {
+    return '*';
+}
+```
+
+  </TabItem>
+  <TabItem value="mimetypes" label="Filter mimetypes" default>
+
+```php
+function supported_filetypes() {
+    return ['image/gif', 'image/jpeg', 'image/png'];
+}
+```
+
+  </TabItem>
+  <TabItem value="groups" label="Filter file groups" default>
+
+```php
+function supported_filetypes() {
+    return ['web_image'];
+}
+```
+
+  </TabItem>
+</Tabs>
+
+For a full list of possible types and groups, look in lib/filelib.php, function get_mimetypes_array().
+
+#### Course and User Repository Instances.
+
+By default, a system wide instance of a repository is created when the repository is enabled.  If you want to allow either or both course specific and user repository instances, you have to specify both **enablecourseinstances** and **enableuserinstances** as options.  There are three ways this can be done:
+
+1. Define **$string\['enablecourseinstances'\]** and **$string\['enableuserinstances'\]** in your plugin's language file.
+2. Add them as items in **get_type_option_names()** function's return array.  Note, you must not define the form fields for these options in the **type_config_form()** function.
+3. Several 'core' repositories use the **db/install.php** to create the original repository instance by constructing an instance of the **repository_type** class.  The options can be defined in the array passed as the second parameter to the constructor.
+
```

Comment:
It would be a good idea to create an example, in tabs (one tab per file), in a collapsed section.